custom_components/ble_monitor/ble_parser/bparasite.py

```diff
@@ -0,0 +1,87 @@
+"""Parser for BParasite BLE advertisements"""
+import logging
+from struct import unpack
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def parse_bparasite(self, data, source_mac, rssi):
+    """Check for adstruc length"""
+    device_type = "b-parasite"
```

Comment:
You will need to add the device type here. The device type can't be the same as the "brand". E.g. for Xiaomi sensors, the device type is e.g. called LYWSD03MMC, not Xiaomi. My proposal

Manufacturer = `B-Parasite`
Device Type = `Plant sensor v1` (up to you to decide, but keep in mind that you might want to create other sensors in the future, so a version number might be useful)
Firmware = `B-Parasite v1` or `B-Parasite v2`

Reply:
The manufacturer in this case would be rbaron, the creator of the device. There are multiple versions of the same device (b-parasite), with and without illuminance (-> Device Types). They are called b-parasite V1.0.0 and b-parasite V1.0.0. There is a third revision, V1.2.0, but that had only hardware changes and does not affect the communication. I have changed this accordingly.

Reply:
Ok, your change makes sense, in that case.